adapters/bidder.go

```diff
@@ -0,0 +1,69 @@
+package adapters
+
+import (
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+// Bidder describes how to connect to external demand.
+type Bidder interface {
+	// MakeRequests makes the HTTP requests which should be made to fetch bids.
+	//
+	// nil return values are acceptable, but nil elements *inside* those slices are not.
+	//
+	// The errors should contain a list of errors which explain why this bidder's bids will be
+	// "subpar" in some way. For example: the request contained ad types which this bidder doesn't support.
+	MakeRequests(request *openrtb.BidRequest) ([]*RequestData, []error)
+
+	// MakeBids unpacks the server's response into Bids.
+	//
+	// The bids can be nil (for no bids), but should not contain nil elements.
+	//
+	// The errors should contain a list of errors which explain why this bidder's bids will be
+	// "subpar" in some way. For example: the server response didn't have the expected format.
+	MakeBids(request *openrtb.BidRequest, response *ResponseData) ([]*TypedBid, []error)
+}
+
+// TypedBid packages the openrtb.Bid with any bidder-specific information that PBS needs to populate an
+// openrtb_ext.ExtBidPrebid.
+//
+// TypedBid.Bid.Ext will become "response.seatbid[i].bid.ext.bidder" in the final OpenRTB response.
+// TypedBid.BidType will become "response.seatbid[i].bid.ext.prebid.type" in the final OpenRTB response.
+type TypedBid struct {
+	Bid     *openrtb.Bid
+	BidType openrtb_ext.BidType
+}
+
+// RequestData and ResponseData exist so that prebid-server core code can implement its "debug" functionality
+// uniformly across all Bidders.
+// It will also let us experiment with valyala/vasthttp vs. net/http without changing every adapter (see #152)
+
+// ResponseData packages together information from the server's http.Response.
+type ResponseData struct {
+	StatusCode int
+	Body       []byte
+	Headers    http.Header
+}
+
+// RequestData packages together the fields needed to make an http.Request.
+type RequestData struct {
```

Comment:
@dbemiller @hhhjort where do you set basic auth for the request now that only the headers are being set?